bundles/tools.vitruv.change.atomic/src/tools/vitruv/change/atomic/uuid/UuidResolverImpl.java

```diff
@@ -0,0 +1,268 @@
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+import tools.vitruv.change.atomic.id.IdResolver;
+
+class UuidResolverImpl implements UuidResolver {
+	static private final Logger LOGGER = Logger.getLogger(UuidResolverImpl.class);
+	private static final String NON_READONLY_PREFIX = "ord_";
+
+	private static final String SERIALIZATION_SEPARATOR = "|";
+
+	private final ResourceSet resourceSet;
+	private final BiMap<EObject, String> eObjectToUuid = HashBiMap.create();
+
+	public UuidResolverImpl(ResourceSet resourceSet) {
+		checkArgument(resourceSet != null, "Resource set may not be null");
+		this.resourceSet = resourceSet;
+	}
+
+	@Override
+	public String getUuid(EObject eObject) throws IllegalStateException {
+		String uuid = getUuidOrNull(eObject);
+		checkState(uuid != null, "no UUID could be found for EObject: %s", eObject);
+		return uuid;
+	}
+
+	@Override
+	public EObject getEObject(String uuid) throws IllegalStateException {
+		EObject eObject = getEObjectOrNull(uuid);
+		checkState(eObject != null, "no EObject could be found for UUID: %s", uuid);
+		return eObject;
+	}
+
+	@Override
+	public void registerEObject(String uuid, EObject eObject) throws IllegalStateException {
+		checkState(uuid != null, "uuid must not be null");
+		checkState(eObject != null, "object must not be null");
+		if (eObject.eResource() != null && eObject.eResource().getResourceSet() != null) {
+			checkState(eObject.eResource().getResourceSet() == resourceSet,
+					"element %s is contained in wrong resource set", eObject);
+		}
+		checkState(eObjectToUuid.getOrDefault(eObject, uuid).equals(uuid),
+				"element %s is already registered for UUID %s", eObject, eObjectToUuid.get(eObject));
+		checkState(eObjectToUuid.inverse().getOrDefault(uuid, eObject).equals(eObject),
+				"UUID %s is already registered for element %s, was trying to register %s", uuid,
+				eObjectToUuid.inverse().get(uuid), eObject);
+		if (isReadOnlyEObject(eObject)) {
+			String expectedUuid = getUuidForReadOnlyEObject(eObject);
+			checkState(uuid.equals(expectedUuid), "read-only object %s must be registered for UUID %s but was %s",
+					eObject, expectedUuid, uuid);
+			return;
+		}
+		if (LOGGER.isTraceEnabled()) {
+			LOGGER.trace("Adding UUID " + uuid + " for EObject: " + eObject);
+		}
+		eObjectToUuid.put(eObject, uuid);
+	}
+
+	@Override
+	public String generateUuid(EObject eObject) {
+		checkState(!eObject.eIsProxy(), "Cannot generate UUID for proxy object %s", eObject);
+		if (isReadOnlyEObject(eObject)) {
+			return getUuidForReadOnlyEObject(eObject);
+		}
+		return NON_READONLY_PREFIX + EcoreUtil.generateUUID();
+	}
+
+	@Override
+	public Resource getResource(URI uri) {
+		return getOrCreateResource(resourceSet, uri);
+	}
+
+	@Override
+	public void endTransaction() {
+		cleanupRemovedElements();
+	}
+
+	@Override
+	public void resolveResources(Map<Resource, Resource> sourceToTargetResourceMapping,
```

Comment:
The ResourceSets are different and both have to cotain resources that belong to their corresponding ResourceSet. Comment from @JanWittler "You need the ResourceSet of the targetUUidResolver and since you can't access it via the private getter, you have to do it this way."

Reply:
👍